src/test/regress/expected/validate_constraint.out

```diff
@@ -0,0 +1,256 @@
+  ADD CONSTRAINT failing_constraint CHECK (constrained_column > 100) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT failing_constraint;
+ERROR:  check constraint "failing_constraint_8000009" is violated by some row
+CONTEXT:  while executing command on localhost:57637
+-- a constraint that can be validated on all workers but one
+ALTER TABLE constrained_table
+  ADD CONSTRAINT failing_on_1_worker CHECK (constrained_column <> 125) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT failing_on_1_worker;
+ERROR:  check constraint "failing_on_1_worker_8000014" is violated by some row
+CONTEXT:  while executing command on localhost:57638
+-- Check which constraints are validated
+SELECT *
+FROM constraint_validations
+ORDER BY 1, 2;
+       Constraint       | Validated? 
+------------------------+------------
+ failing_constraint     | f
+ failing_on_1_worker    | f
+ validatable_constraint | t
+(3 rows)
+
+SELECT *
+FROM constraint_validations_in_workers
+ORDER BY 1, 2;
+              name              | validated 
+--------------------------------+-----------
+ failing_constraint_8000009     | f
+ failing_constraint_8000010     | f
+ failing_constraint_8000011     | f
+ failing_constraint_8000012     | f
+ failing_constraint_8000013     | f
+ failing_constraint_8000014     | f
+ failing_constraint_8000015     | f
+ failing_constraint_8000016     | f
+ failing_on_1_worker_8000009    | f
+ failing_on_1_worker_8000010    | f
+ failing_on_1_worker_8000011    | f
+ failing_on_1_worker_8000012    | f
+ failing_on_1_worker_8000013    | f
+ failing_on_1_worker_8000014    | f
+ failing_on_1_worker_8000015    | f
+ failing_on_1_worker_8000016    | f
+ validatable_constraint_8000009 | t
+ validatable_constraint_8000010 | t
+ validatable_constraint_8000011 | t
+ validatable_constraint_8000012 | t
+ validatable_constraint_8000013 | t
+ validatable_constraint_8000014 | t
+ validatable_constraint_8000015 | t
+ validatable_constraint_8000016 | t
+(24 rows)
+
+DROP TABLE constrained_table;
+CREATE TABLE referenced_table (
+  id          int UNIQUE,
+  test_column int
+);
+SELECT create_distributed_table('referenced_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE referencing_table (
+  id     int,
+  ref_id int
+);
+SELECT create_distributed_table('referencing_table', 'ref_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- inserts some invalid references
+INSERT INTO referencing_table
+SELECT x, x
+from generate_series(1, 100) as f (x);
+ALTER TABLE referencing_table
+  ADD CONSTRAINT fk_constraint FOREIGN KEY (ref_id) REFERENCES referenced_table (id) NOT VALID;
+-- should fail as the foreign keys are not present
+ALTER TABLE referencing_table
+  VALIDATE CONSTRAINT fk_constraint;
+ERROR:  insert or update on table "referencing_table" violates foreign key constraint "fk_constraint"
+DETAIL:  Key (ref_id)=(1) is not present in table "referenced_table".
```

Comment:
Similar here. Validation output may likely to change in some executions and end-up with random test failures, which is not good. 

I think, I can see your goal of this test. You're trying to test whether the constraint is applied afterwards even if it is not valid. Well, that's something PostgreSQL enforces, so I'd be OK to remove these tests. Because, other parts of the tests are already covered by the above tests.

Reply:
After removing the tests that are not needed as they are enforced by Postgres, I arrived at the following:

- insert invalid references
- add not valid constraint
- ~fail to validate~
- ~fail to insert invalid references as Postgres enforces not valid constraints on insertion~
- create missing references
- validate

Now these steps does not really mean much, so I am removing these steps all together